src/python/pants/backend/scala/target_types.py

```diff
@@ -36,9 +37,12 @@ class ScalaSettingsRequest(TargetFilesGeneratorSettingsRequest):
 
 
 @rule
-def scala_settings_request(_: ScalaSettingsRequest) -> TargetFilesGeneratorSettings:
-    # TODO: See https://github.com/pantsbuild/pants/issues/14382.
-    return TargetFilesGeneratorSettings(add_dependencies_on_all_siblings=True)
+def scala_settings_request(
+    scala_infer_subsystem: ScalaInferSubsystem, _: ScalaSettingsRequest
+) -> TargetFilesGeneratorSettings:
+    return TargetFilesGeneratorSettings(
```

Comment:
We want to deprecate if this value is not explicitly set. You'll use `warn_or_error()`, and call it if `scala_infer_subsystem.options.is_default("force_add_siblings_as_dependencies")`.

For the deprecation message, you can probably pluck some of this from above:

> Setting this option to true reduces the precision of dependency information. That means that you may end up compiling more than is necessary for a particular task, and that compilation will be invalidated more frequently than actually necessary. However, setting to true may be helpful if compilation fails due to missing dependencies.

I think we might also want to explain something like:

 > We have made several improvements to Pants's Scala dependency inference, where we no longer think it's necessary to adding dependencies on sibling targets. If you have compilation failures after disabling this option, please consider opening an issue at https://github.com/pantsbuild/pants/issues/new so that we can continue to improve Pants's dependency inference.

Reply:
done